util/osshare/filesharingstatus_windows.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build windows
+
+package osshare
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"golang.org/x/sys/windows/registry"
+)
+
+const (
+	sendFileShellKey = `*\shell\tailscale`
+)
+
+var (
+	muIpnExePath sync.Mutex // locks the following var
+	ipnExePath   string     // lazily initialized absolute path of tailscale-ipn.exe
+)
+
+func getIpnExePath() (string, error) {
+	muIpnExePath.Lock()
+	defer muIpnExePath.Unlock()
+
+	if ipnExePath == "" {
+		// Find the absolute path of tailscale-ipn.exe assuming that it's in the same
+		// directory as this executable (tailscaled.exe).
+		var err error
+		if ipnExePath, err = os.Executable(); err != nil {
+			return "", fmt.Errorf("os.Executable error: %v", err)
+		}
+		if ipnExePath, err = filepath.EvalSymlinks(ipnExePath); err != nil {
+			ipnExePath = ""
+			return "", fmt.Errorf("filepath.EvalSymlinks error: %v", err)
+		}
+		ipnExePath = filepath.Join(filepath.Dir(ipnExePath), "tailscale-ipn.exe")
+		ipnExePath, err = filepath.Abs(ipnExePath)
+		if err != nil {
+			ipnExePath = ""
+			return "", fmt.Errorf("filepath.Abs error: %v", err)
+		}
+	}
+
+	return ipnExePath, nil
+}
+
+// SetFileSharingEnabled adds/removes "Send with Tailscale" from Windows shell menu.
+func SetFileSharingEnabled(enabled bool) error {
```

Comment:
can you also call this from tailscaled's `func uninstallSystemDaemonWindows` ?

Reply:
Sure, I'll add. The key is also removed by the installer on uninstall, but it won't hurt to add it there.